### DISCUSSION TITLE
SHA-256 checksum of a file in datastore

### DISCUSSION THREAD

**Discussion by elungu3** [vmware/pyvmomi-community-samples#751] [Q&A]
Hello team and thanks again for the great work you're doing.

Here is a scenario: I want to upload an iso file to a specific datastore (which I am able to), but I am uploading the file no matter what, which I am going to use to create a vm. Next thing I did was a http GET method to the same url that I upload iso files to, in order to check if that iso file is already there, but that doesn't solve the scenario when 2 iso files have the same name, but they're different files. 

Q: is there a way to do a checksum or something similar for a specific file in a specific datastore in order to compare it with a file I want to upload?

Thank you in advance for your time.

**Comment by smabkhout** [comment]
Hey, it's been some time but I am facing the same problem. I believe the best approach is to just always upload the iso file along with a checksum file. This way, the next time you try to import, you check the previously uploaded checksum file and see if it matches yours locally.
